### DISCUSSION TITLE
Deaktivierung/Dimmen der Onboard-LED

### DISCUSSION THREAD

**Discussion by MrHonk** [tbnobody/OpenDTU#771] [Ideas]
Moin moin,

ich nutze openDTU mit einem ESP Wroom 32 auf einem "ESP32 OpenDTU ADAPTER V1" Board. Damit das Gehäuse passend zu meiner Anbauwand ist, habe ich es mit weißem Filament gedruckt. Das Ergebnis ist für mich auch soweit ganz zufriedenstellend, allerdings stört mich noch ein wenig die rot durch das Gehäuse leuchtende Onboard-LED des ESP32 Boards, weshalb ich eine Funktion in der WebUI begrüßen würde, mit der man die LED entweder komplett deaktiveren, oder aber zumindest in der Helligkeit dimmen könnte.

Meine Programmierkenntnisse reichen hierfür leider nicht aus, aber ich habe im Netz schon gelesen, dass man die LED programmiertechnisch zum Blinken bringen kann, demnach sollte es doch auch möglich sein, diese zumindest komplett zu deaktivieren.

Der Link zur Seite inkl. Code lautet: https://steemit.com/esp32/@armageddit/einfuehrung-2-blinken-der-onbord-led

Vielleicht finden sich ja noch ein paar weitere Interessenten, die das grell rote Licht der Onboard-LED ebenfalls stört und diese gerne deaktivieren oder dimmen können möchten.

LG
Thomas

**Comment by tbnobody** [comment]
Aktuell gibt es keinen LED Support. D.h. die LED die jetzt schon blinkt ist wohl hart verdrahtet und nicht via Source Code steuerbar.

**Comment by MrHonk** [comment]
Hmm, auf der von mir verlinkten Seite ist folgender Code angegeben, der die Onboard-LED angeblich (angeblich, weil ich das nicht prüfen kann) blinkend ansteuern kann:

```
int LED_BUILTIN = 2;

 // the setup function runs once when you press reset or power the board
void setup() {
// initialize digital pin LED_BUILTIN as an output.
pinMode(LED_BUILTIN, OUTPUT);
}

// the loop function runs over and over again forever
void loop() {
  digitalWrite(LED_BUILTIN, HIGH);   // turn the LED on (HIGH is the voltage level)
  delay(1000);                       // wait for a second
  digitalWrite(LED_BUILTIN, LOW);    // turn the LED off by making the voltage LOW
  delay(1000);                       // wait for a second
}
```

Wenn ich das richtig sehe, dann würden dafür diese Zeilen für die Deaktivierung doch ausreichen, oder nicht?
```
int LED_BUILTIN = 2;

 // the setup function runs once when you press reset or power the board
void setup() {
// initialize digital pin LED_BUILTIN as an output.
pinMode(LED_BUILTIN, OUTPUT);
}

void loop() {
  digitalWrite(LED_BUILTIN, LOW);    // turn the LED off by making the voltage LOW;
}
```

LG
Thomas

**Comment by tbnobody** [comment]
Den Source habe ich durchaus gelesen. Schaltet ja nur den Pin High/Low. Das Board hat aber 2 LED's. Eine an einem GPIO, die andere mit Vorwiderstand an +5V. 
Vgl: https://esp32-server.de/wp-content/uploads/ESP32_Schaltplan.jpg

Es gibt auch Boards die haben noch zusätzliche LEDs an den RX/TX Leitungen. Diese blinken dann  eher im Takt der seriellen Konsole.

**Comment by MrHonk** [comment]
Ah okay, da steht ja auch pinMode. Wusste nicht, dass das Board quasi 2 LEDs hat. Danke für das Schaltbild und die Infos.

Edit: Habe die LED jetzt mal mit Panzerband abgeklebt, so scheint sie jedenfalls nicht mehr durch das Gehäuse durch und wenn ich den Deckel abnehme kann ich trotzdem noch sehen ob sie leuchtet oder nicht. Nicht schön, wahrscheinlich auch nicht gerade elegant, aber funktioniert wenigstens :-)

**Reply by 3DJupp** [reply]
@MrHonk  Wenn du die LEDs ablötest oder vorsichtig trennst ist es auch erledigt 😄 
Ich bin Grade daran, eine LED nachzurüsten, ist also immer Ansichtssache ✌️

**Comment by tbnobody** [comment]
Ändere einfach die LED pins in deinem Device Profile auf -1. Damit werden die LEDs deaktiviert.

https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles.md

**Comment by Jens869** [comment]
wenn die LED über einen Widerstand fest an Plus und Minus ist hilft nur auslöten.
wenn diese an einen IO Pin ist kann diese gesteuert werden

**Comment by Marc9485** [comment]
Tesafilm und Edding oder schwarzes Isoband drüber kleben...
